### PR TITLE
Enable decal editing in Linux

### PR DESCRIPTION
**Pull request by JoePfeiffer**
There appear to have been a total of three typos preventing decal editing in Linux
1) The test for snap confinement ended up not allowing any UNIX-like OSes to set the editor in the preferences
2) When the tests for desktop support failed, the text entry field for the command name and the command chooser were not enabled.

### CODE REVIEW

File: swing/src/net/sf/openrocket/gui/dialogs/preferences/GraphicsPreferencesPanel.java [hunk 1]
```diff
@@ -145,7 +145,7 @@ public void stateChanged(ChangeEvent e) {
 		 * we will rely on using the xdg-open command which allows the user to pick
 		 * their preferred application.
 		 */
-		if (SystemInfo.getPlatform() != Platform.UNIX && !SystemInfo.isConfined()) {
+		if (SystemInfo.getPlatform() != Platform.UNIX || !SystemInfo.isConfined()) {
```
Comment (wolsen) [inline swing/src/net/sf/openrocket/gui/dialogs/preferences/GraphicsPreferencesPanel.java:148]: Just looking at this and reading the comment defining the intention, I feel that this should actually be:

```java
if (SystemInfo.getPlaform() == Platform.UNIX && !SystemInfo.IsConfined()) {
...
```

Comment (wolsen) [reply swing/src/net/sf/openrocket/gui/dialogs/preferences/GraphicsPreferencesPanel.java:148]: Scratch that - the logic is correct, it just took me a moment.

Essentially, we want to run it we're not on Linux and its not a confined state. Your's does the trick

**Comment by wolsen** [comment]
This works correctly in the snap packaging. There is another issue where the gimp snap gets rejected access to the decal png - but this bug is unrelated to this change, previously worked, and is likely an issue with the snapd or gimp snap. I'll follow up on those.

**Comment by JoePfeiffer** [comment]
At least under Linux, it isn't necessary for isSupported(Desktop.Action.EDIT)  to be true in order to launch an editor program -- as I understand it, it can't be done from inside a snap, but isn't a problem otherwise.  If the EDIT action is always defined under MacOS and Windows, this should be ready to go.  If not, we'll need another round.

Is there documentation anywhere listing Java's supported desktops and actions, on a per-platform basis?

**Pull request merged**